src/pages/PersonalisationWizardPage.js

```diff
@@ -39,9 +39,10 @@ class PersonalisationWizardPage extends BasePersonalisationPage {
     }
 
     nextSubPage(): ?ReactClass {
-        const nextSubpageIdx = this.currentComponentIdx + 1;
-
-        return this.personalisationComponents[nextSubpageIdx];
+        return this.personalisationComponents.find((component, idx) =>
+            (idx > this.currentComponentIdx) &&
+            (!component.getSearch({}))
+        );
```

Comment:
As discussed - will not work as we need to check the index (which isn't available in a comprehension).